source/renderer/app/components/assets/Asset.tsx

```diff
@@ -173,8 +173,19 @@ class Asset extends Component<Props, State> {
       hasWarning,
       hasError,
     } = this.props;
-    const { fingerprint, metadata, decimals, recommendedDecimals } = asset;
-    const { name } = metadata || {};
+    const {
+      fingerprint,
+      metadata,
+      decimals,
+      recommendedDecimals,
+      assetName,
+    } = asset;
+    const hasName = !!metadata?.name;
+    const name =
+      metadata?.name || (assetName && `ASCII: ${hexToString(assetName)}`);
+    const displayName = metadataNameChars
+      ? ellipsis(name, metadataNameChars)
```

Comment:
In case `assetName` is `undefined` this guy will throw an error. I know that the probability for such case is really low, but maybe it would be worth to secure for such case as the `Asset` data is external so we have no control over it.

Reply:
Yes, I'll update. Thank you!